### PR TITLE
fix logic error

### PR DESCRIPTION
**Pull request by torusrxxx**
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/capstone_wrapper/6)

### CODE REVIEW

File: capstone_wrapper.cpp [hunk 1]
```diff
@@ -513,13 +513,13 @@ bool Capstone::IsBranchGoingToExecute(size_t cflags, size_t ccx) const
     case X86_INS_JE:
         return bZF;
     case X86_INS_JGE:
-        return bSF == bOF;
+        return bSF == bOF || bZF;
```
Comment (mrexodia) [inline capstone_wrapper.cpp]: Refer to the official manual:

jge

http://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf

File: capstone_wrapper.cpp [hunk 2]
```diff
@@ -503,7 +503,7 @@ bool Capstone::IsBranchGoingToExecute(size_t cflags, size_t ccx) const
     case X86_INS_JA:
         return !bCF && !bZF;
     case X86_INS_JBE:
-        return bCF && bZF;
+        return bCF || bZF;
```
Comment (mrexodia) [inline capstone_wrapper.cpp:506]: This is correct.

File: capstone_wrapper.cpp [hunk 3]
```diff
@@ -513,13 +513,13 @@ bool Capstone::IsBranchGoingToExecute(size_t cflags, size_t ccx) const
...
     case X86_INS_JGE:
-        return bSF == bOF;
+        return bSF == bOF || bZF;
     case X86_INS_JG:
         return !bZF && bSF == bOF;
     case X86_INS_JLE:
         return bZF || bSF != bOF;
     case X86_INS_JL:
-        return bSF != bOF;
+        return bSF != bOF && !bZF;
```
Comment (mrexodia) [inline capstone_wrapper.cpp]: This is also not in correspondence with the manual.

File: capstone_wrapper.cpp (hunk 2 above)
Comment (torusrxxx) [reply capstone_wrapper.cpp:506]: ok things are fixed

**Comment by mrexodia** [comment]
Good, I have also updated https://github.com/x64dbg/idaref for the incorrect JBE (I took the logic from that file).

**Pull request merged**